cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,71 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
+import SignUpPage from '../support/pages/signUp.pageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+import faker from 'faker';
 
 describe('Sign Up page', () => {
+  let name;
+  let email;
+  let password;
+  const signUpPage = new SignUpPage();
+
   before(() => {
+    cy.task('db:clear');
+  });
+
+  beforeEach(() => {
+    cy.visit('/#/register');
+    name = faker.name.firstName();
+    email = faker.internet.email();
+    password = faker.internet.password();
+  });
+
+  it('should create a new user', () => {
+    signUpPage.fillName(name);
+    signUpPage.fillEmail(email);
+    signUpPage.fillPassword(password);
+
+    signUpPage.submitSignUpForm();
+
+    cy.contains('Welcome!');
+    cy.contains('Your registration was successful!');
+    cy.contains('OK').click();
+
+    const homePage = new HomePageObject();
+    homePage.assertHeaderContainUsername(name);
+  });
+
+  it('should see warning message for empty username field', () => {
```

Comment:
You're creating users with missing fields to check validation messages, but you're not checking the case where a user tries to sign up with credentials that are already in use. Consider adding a test for this scenario.

Reply:
done

Reply:
done

Reply:
done